flows/events/time_wait.go

```diff
@@ -0,0 +1,35 @@
+package events
+
+import "github.com/nyaruka/goflow/flows"
+
+// TypeTimeWait is the type of our time wait event
+const TypeTimeWait string = "time_wait"
+
+// TimeWaitEvent events are created when a flow requests to be paused for a certain amount of time.
+// ```
+//   {
+//     "type": "time_wait",
+//     "created_on": "2006-01-02T15:04:05Z",
+//     "timeout": 300
+//   }
+// ```
+//
+// @event time_wait
+type TimeWaitEvent struct {
```

Comment:
This isn't used anywhere yet but I think it's a good example of why having waits check run state rather than match against a specific event, makes sense. This wait doesn't check anything and could be used with timeout=0 to have a caller commit the current session and immediately call back.